### PR TITLE
Ha low enum additions

### PR DESCRIPTION
**Pull request by simonhmorris1**
This PR adds the HaLow enum for WiFiBand and WiFiVersionEnum as described here: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6333#pullrequestreview-1323895640

### CODE REVIEW

File: src/darwin/Framework/CHIP/templates/availability.yaml
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/templates/availability.yaml:1]: I appreciate you trying to fix these bits, but the changes to this file aren't right.  They are claiming that these new values shipped in iOS 16.1 for "1G" and iOS 16.4 for "Ah", which is hard without time-travel.

Please just revert the changes to this file and I'll update it after this PR lands.  Or I can push an additional changeset to this PR, or a PR against this PR, to put the right annotations in.  Let me know which of those you would prefer, @simonhmorris1

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
@simonhmorris1 Note also the restyled failures in the C++ changes.

File: src/darwin/Framework/CHIP/templates/availability.yaml
Comment (simonhmorris1) [reply src/darwin/Framework/CHIP/templates/availability.yaml:1]: Reverted the availability change, thanks for pointing this out. Please feel free to add a changeset to this PR. The auto generated  MTRBaseClusters.h now shows MTR_NEWLY_AVAILABLE. @bzbarsky-apple

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@simonhmorris1 You need to merge/rebase to tip and run `/scripts/tools/zap/generate.py examples/chef/devices/rootnode_pump_a811bb33a0.zap` to regenerate the bits for the new example app that got added...

**Review by andy31415 — APPROVED** [review]

**Pull request merged**